editors/sc-ide/core/main.cpp

```diff
@@ -135,22 +131,7 @@ int main( int argc, char *argv[] )
     if (startInterpreter)
         main->scProcess()->startLanguage();
 
-    // setup HelpBrowser server
-    QWebSocketServer server("SCIDE HelpBrowser Server", QWebSocketServer::NonSecureMode);
-    if (!server.listen(QHostAddress::LocalHost, 12344)) {
-        qFatal("Failed to open web socket server.");
-        return 1;
-    }
-
-    // setup comm channel
-    WebSocketClientWrapper clientWrapper(&server);
-    QWebChannel channel;
-    QObject::connect(&clientWrapper, &WebSocketClientWrapper::clientConnected,
-                     &channel, &QWebChannel::connectTo);
-
-    // publish IDE interface
-    IDEWebChannelWrapper ideWrapper{win->helpBrowserDocklet()->browser()};
-    channel.registerObject("IDE", &ideWrapper);
+    win->helpBrowserDocklet()->browser()->setUpServer();
```

Comment:
i've tried calling `setUpServer()` in the constructor for `HelpBrowser`, but it just doesn't work (client does not establish connection and prints an error). it has to be called in this exact spot. i'm not sure why

Reply:
you have a dangling pointer in `setUpServer`. that might have something to do with it; I'll try to review this PR in detail soon (this week). thanks for getting it started!